src/main/java/com/datadog/api/client/ApiClient.java

```diff
@@ -1562,6 +1582,31 @@ public <T> ApiResponse<T> invokeAPI(
     }
   }
 
+  private boolean shouldRetry(int retryCount, int statusCode, RetryConfig retryConfig) {
+    boolean statusToRetry = false;
+    if (statusCode == 429 || statusCode >= 500) {
+      statusToRetry = true;
+    }
+    return (retryConfig.maxRetries >= retryCount && statusToRetry && retryConfig.isEnableRetry());
+  }
+
+  private int calculateRetryIntrval(
+      Map<String, List<String>> responseHeaders, RetryConfig retryConfig, int retryCount) {
+    if (responseHeaders.get("x-ratelimit-reset") != null) {
+      List<String> rateLimitHeader = responseHeaders.get("x-ratelimit-reset");
+      String ratelimitString = rateLimitHeader.get(0);
+      String ratelimitStringTrimmed = ratelimitString.replaceAll("\\D", "");
```

Comment:
Why do you need that?

Reply:
The header is a list of strings because one header can have multiple value. And `rateLimitHeader.get(0);` returns a string `[1]`